presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/file/FileHiveMetastore.java

```diff
@@ -826,7 +836,7 @@ public synchronized void revokeRoles(Set<String> roles, Set<HivePrincipal> grant
         }
         result.addAll(listRoleGrantsSanitized().stream()
                 .filter(grant -> HivePrincipal.from(grant.getGrantee()).equals(principal))
-                .collect(toSet()));
+                .collect(toUnmodifiableSet()));
```

Comment:
This is unrelated to the change so it should go as separate commit. 

Instead `collect` and `addAll` I would use `forEach(result::add)`